bridge/client/app/_views/ktb-settings-view/ktb-project-settings/ktb-project-settings.component.ts

```diff
@@ -68,22 +66,95 @@ export class KtbProjectSettingsComponent implements OnInit, OnDestroy, PendingCh
 
   public message = 'You have pending changes. Make sure to save your data before you continue.';
   public unsavedDialogState: DialogState = null;
-  public resourceServiceEnabled?: boolean;
+
+  public resourceServiceEnabled$ = this.featureFlagService.featureFlags$.pipe(
+    map((featureFlags: IClientFeatureFlags) => featureFlags.RESOURCE_SERVICE_ENABLED)
+  );
+
+  public projectName$: Observable<string | null> = this.route.paramMap.pipe(
+    map((params) => params.get('projectName')),
+    tap((projectName) => {
+      if (projectName) {
+        this.projectName = projectName;
+        this.projectDeletionData = {
+          type: DeleteType.PROJECT,
+          name: projectName,
+        };
+      }
+    })
+  );
+
+  public gitInputDataExtended$: Observable<IGitDataExtended | undefined> = this.projectName$.pipe(
+    mergeMap((projectName) => (projectName ? this.dataService.loadPlainProject(projectName) : of(undefined))),
+    map((project) => project?.gitCredentials)
+  );
+
+  public projectNames$: Observable<string[] | undefined> = this.projectName$.pipe(
+    mergeMap((projectName) => (!projectName ? this.dataService.projects : of(undefined))),
+    map((projects: Project[] | undefined) => projects?.map((project) => project.projectName))
+  );
+
+  public projectCreated$: Observable<boolean> = this.route.queryParams.pipe(map((queryParams) => queryParams.created));
 
   readonly state$: Observable<ProjectSettingsState> = combineLatest([
     this.dataService.keptnInfo,
     this.dataService.keptnMetadata,
+    this.projectName$,
+    this.resourceServiceEnabled$,
+    this.gitInputDataExtended$,
+    this.projectNames$,
+    this.projectCreated$,
   ]).pipe(
-    filter((info): info is [KeptnInfo, IMetadata | undefined | null] => !!info[0]),
-    map(([keptnInfo, metadata]) => {
-      return {
-        gitUpstreamRequired: !metadata?.automaticprovisioning,
-        automaticProvisioningMessage: keptnInfo.bridgeInfo.automaticProvisioningMsg,
-        state: metadata === null ? ProjectSettingsStatus.ERROR : ProjectSettingsStatus.LOADED,
-      };
-    }),
+    filter(
+      (
+        info
+      ): info is [
+        KeptnInfo,
+        IMetadata | undefined | null,
+        string,
+        boolean,
+        IGitDataExtended | undefined,
+        string[] | undefined,
+        boolean
+      ] => !!info[0]
+    ),
```

Comment:
This filter is completely wrong. The types of keptnInfo and projectName will be incorrectly changed here. After the filter, the keptnInfo may still be undefined and projectName may be null.
This would solve it (without the filter)
```ts
readonly state$: Observable<ProjectSettingsState> = combineLatest([
    this.dataService.keptnInfo.pipe(filter((info): info is KeptnInfo => !!info)),
    this.dataService.keptnMetadata,
    this.projectName$.pipe(map((projectName) => projectName ?? undefined)),
    this.resourceServiceEnabled$,
    this.gitInputDataExtended$,
    this.projectNames$,
    this.projectCreated$,
  ]).pipe(
...
```

Reply:
Right, the projectName may be null. But the filter for keptnInfo should work. Anyway your suggestion is much more readable. 👍🏻